### PR TITLE
Add encoder coalescing for native kernels

### PR DESCRIPTION
**Pull request by DenisVieriu97**

### CODE REVIEW

**Comment by ssaladis** [comment]
This looks like for Blits to take affect, you are submitting workload to GPU and synchronizing on it for the results. Is there a better way to do this ? Did you look into Barriers/Fences if they would help here ?

File: aten/src/ATen/native/mps/operations/LinearAlgebra.mm [hunk 1]
```diff
@@ -770,6 +770,7 @@ Tensor addbmm_mps(const Tensor& self, const Tensor& batch1, const Tensor& batch2
 
   dispatch_sync(mpsStream->queue(), ^(){
     @autoreleasepool {
+      mpsStream->endKernelCoalescing();
```
[the remark this answers is not in the document]
Comment (kulinseth) [reply aten/src/ATen/native/mps/operations/LinearAlgebra.mm:773]: I think we need to figure out a Global way to track all the kernels.

File: aten/src/ATen/mps/MPSStream.mm [hunk 2]
```diff
@@ -122,6 +139,7 @@
   if (length == 0) return;
   dispatch_sync(_serialQueue, ^() {
     @autoreleasepool {
+      endKernelCoalescing();
```
Comment (kulinseth) [inline aten/src/ATen/mps/MPSStream.mm:142]: I think we should create a similar BlitCommandEncoder coalescing

**Review by kulinseth — APPROVED** [review]
Overall change looks good. Thanks

**Pull request merged**